File: guests/ocaml/grow.ml
(* The grow scene, OCaml port — guests/rust/grow.rs, tools/scenes/grow.steps. *)

open Kaya_app

let () =
  let app = Kaya_app.create () in

  build app (fun () ->
     let probe = signal Scalar.Str ("grow probe") in
     let one = signal Scalar.Str ("one") in

     let root =
       column
         [
           label ~grow:1.0 ~bind:probe (* label#0 *);
           textarea ~grow:2.0 (* textarea#0 *);
           row ~grow:1.0 ~spacing:12.0
             [
               label ~grow:1.0 ~bind:one (* label#1 *);
               button ~grow:3.0 ~text:"three";
             ];
         ]
         ()
     in
     mount root);

  exit (run app)

File: guests/swift/grow.swift
// The grow scene, Swift port — guests/rust/grow.rs, tools/scenes/grow.steps.

import Foundation
import Kaya

KayaApp.run { app in
    app.build { tx in
        let probe = tx.signal(.str("grow probe"))
        let one = tx.signal(.str("one"))

        let root = tx.column { root in
            tx.label(bind: probe, grow: 1)  // label#0
            tx.textarea(grow: 2)  // textarea#0
            tx.row(grow: 1, spacing: 12) { _ in
                tx.label(bind: one, grow: 1)  // label#1
                tx.button("three", grow: 3)
            }
            return root
        }
        tx.mount(root)
    }
}
